internal/dag/gatewayapi_processor.go

```diff
@@ -379,7 +379,7 @@ func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1alpha1.HTTPRo
 
 				if listenerSecret != nil {
 					svhost := p.dag.EnsureSecureVirtualHost(ListenerName{Name: host, ListenerName: "ingress_https"})
-					svhost.Secret = listenerSecret
+					svhost.Secrets = append(svhost.Secrets, listenerSecret)
```

Comment:
The gateway API `featuretests` are failing b/c with multiple HTTPRoute objects or a single one with multiple path matches, this will be called multiple times for the same secret

see https://github.com/projectcontour/contour/blob/efecfc2b8a56a9744b589dedf1e8c844b3660776/internal/dag/gatewayapi_processor.go#L367-L383